File: adventofcode.com/2024/3/2.py
import os
import re

result = 0
enabled = True

with open(os.path.dirname(__file__) + '/input.txt', mode='r') as f:
    for raw_line in f.readlines():
        for operation in re.findall(r'mul\(\d+,\d+\)|do\(\)|don\'t\(\)', raw_line):
            if operation == 'do()':
                enabled = True
            elif operation == 'don\'t()':
                enabled = False
            else:
                if enabled:
                    values = operation[4:-1].split(',')
                    result += int(values[0]) * int(values[1])

print(result)  # 82868252

File: adventofcode.com/2024/03/2.rb
result = 0
enabled = true

File.open(File.dirname(__FILE__) + "/input.txt", "r") do |f|
  f.each_line do |line|
    line.scan(/mul\(\d+,\d+\)|do\(\)|don't\(\)/).each do |operation|
      if operation == "do()"
        enabled = true
      elsif operation == "don't()"
        enabled = false
      else
        if enabled
          values = operation[4..-2].split(',')
          result += Integer(values[0]) * Integer(values[1])
        end
      end
    end
  end
end

puts(result)

